File: boj/02606.py
n = int(input())

g = {k: []for k in range(1, n+1)}
for _ in range(int(input())):
    i, j = map(int, input().split())
    g[i] += [j]
    g[j] += [i]

s, v = [1], [0 for _ in range(n+1)]
while s:
    i = s.pop()
    v[i] = 0
    for j in g[i]:
        if v[j]-1:
            s += [j]

print(sum(v))

File: BOJ/02606.js
const [N, edgeSize, ...input] = require("fs")
  .readFileSync(process.platform === "linux" ? 0 : "./input", "utf-8")
  .trim()
  .split("\n");

const graph = Array(Number(N) + 1)
  .fill(null)
  .map(() => []);

input.forEach((row) => {
  const [a, b] = row.split(" ").map(Number);
  graph[a].push(b);
  graph[b].push(a);
});

const stack = [1];
const visited = new Set([1]);
while (stack.length) {
  const node = stack.pop();
  graph[node].forEach((go) => {
    if (visited.has(go)) return;

    stack.push(go);
    visited.add(go);
  });
}

console.log(visited.size - 1);
